src/filter/rgbsplit0r/rgbsplit0r.c

```diff
@@ -0,0 +1,227 @@
+/* rgbsplit0r.c
+ * Copyright (C) 2016 IDENT Software ~ http://identsoft.org
+ * Inspired by the witch house and web culture
+ * 
+ * This file is a Frei0r plugin.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+ */
+
+#include <stdlib.h>
+#include <assert.h>
+
+#include "frei0r.h"
+
+typedef struct rgbsplit0r_instance
+{
+    unsigned int width;
+    unsigned int height;
+    unsigned int shiftX;
+    unsigned int shiftY;
+
+    uint32_t pxR, pxG, pxB;
+} rgbsplit0r_instance_t;
+
+
+inline static void rgbsplit0r_extract_color(uint32_t *pixelIn, uint32_t *pixelOut,
+                                            short int colorIndex)
+{
+    uint8_t *pxIn  = (uint8_t *)pixelIn,
+            *pxOut = (uint8_t *)pixelOut;
```

Comment:
Tip: you can simply declare the function params as uint8_t* and then you do not need to cast to additional local vars.